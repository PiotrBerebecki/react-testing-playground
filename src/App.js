import React, { Component } from 'react';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="container">
        Welcome to testing
        <ul>
          <li>First</li>
          <li>Second</li>
        </ul>
      </div>
    );
  }
}

export default App;
