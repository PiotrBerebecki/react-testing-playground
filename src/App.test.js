import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import App from './App';

describe('App', () => {
  // "smoke test" - verify that component renders without crashing
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
  });

  // "smoke test" using enzyme's shallow render
  it("renders without crashing (using enzyme's shallow rendering", () => {
    shallow(<App />);
  });

  // DOM testing (manipulate renderered component)
  it("renders without crashing (using enzyme's shallow rendering", () => {
    const component = shallow(<App />);

    component.find('button').simulate('click');
    component.find('button').simulate('click');
    component.find('button').simulate('click');
    // console.log('==========', component.find('.parent__display').text());
    expect(component.find('.parent__display').text()).toContain('Count');
  });

  // snapshot test
  it('renders component correctly', () => {
    const component = renderer.create(<App />);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger callback on button
    tree.children[1].props.onClick();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
