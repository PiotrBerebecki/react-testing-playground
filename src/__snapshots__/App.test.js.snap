exports[`App renders component correctly 1`] = `
<div
  className="container">
  Welcome to testing
  <button
    onClick={[Function]}>
    Click me :)
  </button>
  <div
    className="count">
    Current count: 
    0
  </div>
</div>
`;

exports[`App renders component correctly 2`] = `
<div
  className="container">
  Welcome to testing
  <button
    onClick={[Function]}>
    Click me :)
  </button>
  <div
    className="count">
    Current count: 
    1
  </div>
</div>
`;
